### PR TITLE
refactor(cbr/vault): refactor vault resoruces structure

### LINKED ISSUE #1208: Not being able to attach multiple evs disks to one vault on terraform in an easier way
### Terraform Version

```
Terraform v0.15.0
huaweicloud/huaweicloud v1.25.0
```

### Terraform Configuration Files

```hcl
Today the only code that worked for me is this (**which is a VERY POOR implementation**):

resource "huaweicloud_cbr_vault" "vault" {
  name                   = "vault-${var.env}-${var.project}"
  type                   = var.vault_type
  protection_type        = var.protection_type
  consistent_level       = var.consistent_level
  size                   = var.vault_size
  auto_expand            = var.auto_expand
  policy_id              = var.policy_id

   resources {
     id = "7d95d51b-95a7-4feb-99f9-a81274f35fd1"
     include_volumes  = ["750991e7-9d47-437a-baca-1b2d1323ad42"]
   }
}
```

In my case, I have some instances with just one disk, so I have to put one of these disks as ID and rest of disks IDs (from others machines) in include_volumes filed.

### Expected Behavior

I can think in 2 good solutions for that:

1. Be able to use "resources" ID field as a list (similar to include_volumes), so I can declare multiple disks IDs at once (or multiple instances IDs if I'm using server type), like this:
```hcl
 resources {
     id = "[750991e7-9d47-437a-baca-1b2d1323ad42 , 7d95d51b-95a7-4feb-99f9-a81274f35fd1]"
   }
```
2. Have another way to attach evs disks to one vault (like eip attach "huaweicloud_compute_floatingip_associate_v2", and volume attach "huaweicloud_compute_volume_attach")

### Actual Behavior

Not being able to implement a code that makes sense in terraform, once I dont have an easier way to attach evs disks to a vault. 
If we use hardcoded evs disks ID does not seems so bad, but thinking about automation and terraform it's really a pain in the ass the current implementation.

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The current `resources` structure is not conducive to writing terraform scripts for mounting storage resources (EVS Disk and SFS Turbo).
- `include_volume` and `exclude_volumes` is not used in the update function.
- It is more reasonable to mount multiple disk by list structure.

I adjust the resources structure:
1. `id` to `server_id`: now, the server_id only used to specify ecs instance that attach to vault.
2. `include_volumes` to `includes`: now, the `includes` is used to configure the volume IDs to attach to `disk` and `turbo` vault.
3. `exclude_volumes` to `excludes`: now, the `excludes` is used to specify the disks included in the backup except for the array that contains one or more disk IDs.
4. remove `name`, `protect_status`, `size`, `backup_size` and `backup_count`.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1208

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor include and exclude structure for disk and turbo.
2. adjust parameter name.
3. remove some expired parameters.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCBRV3Vault_BasicServer'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCBRV3Vault_BasicServer -timeout 360m -parallel 4
=== RUN   TestAccCBRV3Vault_BasicServer
=== PAUSE TestAccCBRV3Vault_BasicServer
=== CONT  TestAccCBRV3Vault_BasicServer
--- PASS: TestAccCBRV3Vault_BasicServer (412.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       412.698s
```
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCBRV3Vault_ReplicaServer'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCBRV3Vault_ReplicaServer -timeout 360m -parallel 4
=== RUN   TestAccCBRV3Vault_ReplicaServer
=== PAUSE TestAccCBRV3Vault_ReplicaServer
=== CONT  TestAccCBRV3Vault_ReplicaServer
--- PASS: TestAccCBRV3Vault_ReplicaServer (44.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       44.954s
```
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCBRV3Vault_BasicVolume'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCBRV3Vault_BasicVolume -timeout 360m -parallel 4
=== RUN   TestAccCBRV3Vault_BasicVolume
=== PAUSE TestAccCBRV3Vault_BasicVolume
=== CONT  TestAccCBRV3Vault_BasicVolume
--- PASS: TestAccCBRV3Vault_BasicVolume (108.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       108.401s
```
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCBRV3Vault_BasicTurbo'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCBRV3Vault_BasicTurbo -timeout 360m -parallel 4
=== RUN   TestAccCBRV3Vault_BasicTurbo
=== PAUSE TestAccCBRV3Vault_BasicTurbo
=== CONT  TestAccCBRV3Vault_BasicTurbo
--- PASS: TestAccCBRV3Vault_BasicTurbo (790.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr 790.413s
```
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccCBRV3Vault_ReplicaTurbo'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccCBRV3Vault_ReplicaTurbo -timeout 360m -parallel 4
=== RUN   TestAccCBRV3Vault_ReplicaTurbo
=== PAUSE TestAccCBRV3Vault_ReplicaTurbo
=== CONT  TestAccCBRV3Vault_ReplicaTurbo
--- PASS: TestAccCBRV3Vault_ReplicaTurbo (43.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       43.920s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go [hunk 1]
```diff
@@ -97,41 +125,56 @@ func TestAccCBRV3Vault_serverReplication(t *testing.T) {
...
 func TestAccCBRV3Vault_volumeBasic(t *testing.T) {
 	var vault vaults.Vault
-	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	randName := acceptance.RandomAccResourceName()
 	resourceName := "huaweicloud_cbr_vault.test"
 
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&vault,
+		func(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go]: same code , it is recommended to extract into a method

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+	d.SetId(vault.ID)
+
+	if policy, ok := d.GetOk("policy_id"); ok {
+		_, err := vaults.BindPolicy(client, d.Id(), vaults.BindPolicyOpts{PolicyID: policy.(string)}).Extract()
+		if err != nil {
+			return fmtp.Errorf("Error binding policy to vault: %s", err)
+		}
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
```
Comment (hmbig2) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: The tags package already provides a common method to create tags, which can be used later.

File: huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go]: done

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: done

File: huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go [hunk 3]
```diff
@@ -1,59 +1,74 @@
...
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cbr"
 )
 
-func TestAccCBRV3Vault_serverBasic(t *testing.T) {
+func getResourceObj(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	c, err := conf.CbrV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating HuaweiCloud WAF client: %s", err)
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go]: s/WAF/CBR or remove "WAF"

Comment (Lance52259) [reply huaweicloud/services/acceptance/cbr/resource_huaweicloud_cbr_vault_test.go]: CBR is used

File: docs/resources/cbr_vault.md [hunk 4]
```diff
@@ -103,42 +115,49 @@ The following arguments are supported:
...
+* `resources` - (Optional, List) Specifies an array of one or more resources to attach to the CBR vault.
+  The [object](#cbr_vault_resources) structure is documented below.
 
 * `tags` - (Optional, Map) Specifies the key/value pairs to associate with the CBR vault.
 
+<a name="cbr_vault_resources"></a>
 The `resources` block supports:
 
-* `id` - (Required, String) Specifies the ID of the resource to be backed up.
+* `id` - (Optional, String) Specifies the ID of the resource to be backed up.
```
Comment (niuzhenguo) [inline docs/resources/cbr_vault.md]: s/id/server_id

File: docs/resources/cbr_vault.md [hunk 5]
```diff
@@ -103,42 +115,49 @@ The following arguments are supported:
...
 
-* `exclude_volumes` - (Optional, List) Specifies the disks included in the backup except for the array that contains one
-  or more disk IDs.
+* `excludes` - (Optional, List) Specifies the disks included in the backup except for the array that contains one or
+  more disk IDs. Only **server** vault support this parameter.
 
-* `include_volumes` - (Optional, List) Specifies the array of disk IDs which will be included in the backup. This
-  parameter and `exclude_volumes` are alternative.
+* `includes` - (Optional, List) Specifies the array of disk or SFS file system IDs which will be included in the backup.
```
Comment (niuzhenguo) [inline docs/resources/cbr_vault.md:159]: why `server` vault can't support this?

File: docs/resources/cbr_vault.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/cbr_vault.md]: The element name and description have been updated.

File: docs/resources/cbr_vault.md (hunk 5 above)
Comment (Lance52259) [reply docs/resources/cbr_vault.md:159]: For API limit, the `includes` (whitelist) only provided for `disk` and `sfs turbo`.

File: docs/resources/cbr_vault.md [hunk 6]
```diff
@@ -103,42 +115,49 @@ The following arguments are supported:
...
 
+<a name="cbr_vault_resources"></a>
 The `resources` block supports:
 
-* `id` - (Required, String) Specifies the ID of the resource to be backed up.
+* `server_id` - (Optional, String) Specifies the ID of the ECS instance to be backed up.
 
-* `exclude_volumes` - (Optional, List) Specifies the disks included in the backup except for the array that contains one
-  or more disk IDs.
+* `excludes` - (Optional, List) Specifies the disks included in the backup to which ECS instance belongs except for the
```
Comment (niuzhenguo) [inline docs/resources/cbr_vault.md]: "Specifies the array of disk IDs which will be excluded in the backup." seems more concise.

File: docs/resources/cbr_vault.md (hunk 5 above)
Comment (niuzhenguo) [reply docs/resources/cbr_vault.md:159]: ok

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+			},
+			"resources": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"server_id": {
+							Type:     schema.TypeString,
+							Optional: true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:117]: add `ConflictsWith` field?

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+						},
+						"excludes": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:122]: ditto

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 7 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:117]: Nested maps, sets and lists (greater than one length). do not support `ConflictsWith`.

Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:117]: https://github.com/hashicorp/terraform-plugin-sdk/issues/71

Comment (ShiChangkuo) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go:117]: Ohh, I forgot it.

**Pull request merged**